core/modules/file/views/views_handler_field_file_uri.inc

```diff
@@ -43,12 +43,15 @@ class views_handler_field_file_uri extends views_handler_field_file {
     if (!empty($this->options['file_download_path']) && $data !== NULL && $data !== '') {
       $data = file_create_url($data);
       if (!empty($this->options['image_style'])) {
-      // $data contains url of image.
-      // Find insertion position of '/files'.
-        $insert_at = strpos($data, '/files');
-      // Text to insert in order to modify url.
+        // $data contains url of image.
+        // Get public file system path and its length.
+        $file_public_path = '/' . config_get('system.core','file_public_path');
+        $file_public_path_strlen = strlen($file_public_path);
```

Comment:
@Graham-72 I was also surprised to find that `strlen` returns the number of bytes rather than the number of characters in a string. So in some languages it might return a different result than what is expected: http://php.net/manual/en/function.strlen.php#118484

I realise that usually people will name their directories using English characters, but there might be an odd case where they don't. AFAIK strings are treated as arrays, so you can use `count()` on them. Might wanna consider that 😉

Reply:
@klonos this is interesting and a bit worrying. This render function makes use of php `substr_replace()` in combination with `strlen()`, the latter to provide the correct insertion point for the revised file URL. If both these PHP functions calculate numbers of bytes then perhaps the right answer is obtained. But if `substr_replace() `calculates the start position in characters whereas `strlen()` provides the number of bytes we would seem to have a mismatch.

And I am not sure that utf8_decode() would solve this dilemma. 

Maybe we need to test with 'abnormal' values for the public file system path?

Reply:
I have now done a simple test by modifying the public file system path in a test installation, by adding the character é to the name (this being a two byte character). The `strlen()` function does indeed count bytes rather than characters but `substr_replace()` would seem to be doing the same and the image library is showing the thumbnail correctly.
So there does not seem to be a problem.

Reply:
Good on ya for testing this @Graham-72 👍

Reply:
Could this be rewritten to use `backdrop_strpos` and `backdrop_substr()`? Though we don't have a `backdrop_substr_replace()`, seems as though we could make do. Those functions count the number of characters, not the number of bytes, so they act more intuitively than the PHP native versions. However... if we're getting the right result perhaps we don't have a problem.

Reply:
I think it is OK as it is, using the PHP native functions, since they seem to be consistent with each other, and one as it were cancels the other. I think we would have a problem if we just used backdrop_substr() because we would be mixing counting in characters with counting in bytes.